papers/whitepaper.md

```diff
@@ -111,3 +113,141 @@ The focus of the current text is to explore at a deeper level the Atoma's unique
+
+2. *Smart contract requests:* These consist of any requests that are submitted to the Atoma Network by another smart contract, or that is authenticated by a smart contract. In the first case, a smart contract on the blockchain can make calls directly to the Atoma on-chain logic contract, through external smart contract calls. For example, an on-chain funds manager smart contract can call the Atoma on-chain logic contract to request to have access to an AI prediction of the a given token emission rate for the next 24 hours, based on on-chain market data. In this case, the input request is submitted directly by the smart contract directly to the Atoma on-chain logic contract. This means that the data in fully public and must fit into the blockchain's execution runtime calldata memory limit. Moreover, it is also likely that the smart contract requesting AI inference will need the output to be submitted back on-chain (think of the prediction market case above). However, there are situations in which this might not be the case, for example an on-chain NFT AI art generation protocol. 
+
+3. *Smart contract authenticated requests:* The second case, is the one in which the request is not started by a smart contract, but rather by a wallet, but it requires a smart contract to authenticate the request. Such examples, will include DAO governance decision making through AI, in which a transaction might be submitted through a multi-sig, but it requires approval directly from the majority of the DAO community, etc. Another interesting case is using account abstract 'wallet's to interact with the Atoma Network, in which one can leverage user-defined permission logic with decision making through AI (for example, an AI is required to parse the web to get some data information, and if that data satisfies some well defined logical criteria, blockchain transactions can be submitted). The latter case is especially interesting in the context of the autonomous Smart agents in the context of Web3 protocols. 
+Notice that, in such case the request input data needs not be on-chain data, or either public data and it can be submitted through a frontend UI, a desktop application, a mobile application, similar to the `Externally owned request` case. In this case, the transaction only goes through if the required smart contract 'authorizes' it through the Atoma on-chain logic contract. 
```

Comment:
Notice that, in this case the request input data needs not be on-chain data or public data, and it can be submitted through a frontend UI, desktop application, mobile application, similar to the `Externally owned request` case. In this case, the transaction only goes through if the required smart contract 'authorizes' it through the Atoma on-chain logic contract.